[JW8-11920] Revert enter handling, implement backwards compatibility for deprecated registers

zetagame opened the pull request:
### This PR will...
- Ensure enter is handled on div elements by reverting to the old enter handler pattern.
- Add `type` attribute to menu category buttons to prevent click from being dispatched on enter.

### Why is this Pull Request needed?
Click is only dispatched by enter on button elements, and most interactive UI in the player are div elements.

### Are there any points in the code the reviewer needs to double check?

### Are there any Pull Requests open in other repos which need to be merged with this?

#### Addresses Issue(s):

JW8-11920

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1892/) MULTI Build for commit 0b7807fc96fccc5848cbbf01cd2ad4cf85c6b99b is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9314/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9314/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6259/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6259/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28382/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28382/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23835/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23835/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6003/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28382/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28382/archive/test/view/) Views

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1893/) MULTI Build for commit dbc5b054633977fe084a5658f32488c23e13b313 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9315/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9315/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6260/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6260/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28383/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28383/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23836/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23836/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6004/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28383/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28383/archive/test/view/) Views

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1896/) MULTI Build for commit a789c73d47727f788cbf53c38286ee585bc3e405 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9318/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9318/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6263/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6263/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28386/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28386/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23839/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23839/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6006/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28386/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28386/archive/test/view/) Views

robwalch commented on src/js/view/controls/controlbar.js:
```diff
@@ -334,10 +334,9 @@ export default class Controlbar {
-                // controlbar cast button needs to manually trigger a click
-                // on the native cast button for taps and enter key
-                if (evt.sourceEvent.detail !== 1) {
+            const castUi = elements.cast.ui.on('click enter', function(evt) {
```
Can we use `addClickAction` here?

zetagame replied:
The UI gets created by the `createCastButton` function, which is called in a scope that doesn't have access to the values in this closure.

robwalch commented on src/js/view/controls/controls.js:
```diff
@@ -117,7 +117,7 @@ export default class Controls extends Events {
             const displayContainer = new DisplayContainer(model, api);
 
-            displayContainer.buttons.display.on('click', () => {
+            displayContainer.buttons.display.on('click enter', () => {
```
Can we use `addClickAction` here?

Also UI instances should always be returned and stored as a property of their parent instance (Controls). This is so that we destroy the UI instance when destroying the parent to remove all listeners.

zetagame replied:
Same as above. These are destroyed when `displayContainer.destroy` is called:

<img width="415" alt="image" src="https://user-images.githubusercontent.com/2903955/121192942-06543000-c833-11eb-9084-68c219d875fb.png">

jwrobot-player commented:
[:exclamation:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1897/) MULTI Build for commit 156dccb97af86e0bd999dbc0a8fdd74cdd7dd24c did not complete (FAILURE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9319/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9319/consoleFull) jwplayer build FAILURE
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6264/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6264/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/consoleFull) jwplayer-commercial build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/none/consoleFull) jwplayer-commercial unit tests NOT_STARTED
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/none/allure/) NOT_STARTED
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/view/) Views

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1898/) MULTI Build for commit 9b157be25ddc9614129d7bdba18ab50042a6f5a0 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9320/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9320/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6265/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6265/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28389/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28389/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23842/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23842/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6009/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28389/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28389/archive/test/view/) Views

robwalch commented:
1. Logo click-through is broken:
    - https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6009/allure/#suites/b9cb8385c37c0d6570a24dbf08ce345d/268763c48f174b2d/
    - http://player-pr-test-jenkins.longtailvideo.com/builds/28389/archive/test/harness.html?setup&edition=enterprise&setup&config=player_ui/logo-control-bar-svg.json
2. Found and fixed a regression in the test results with playlist loading https://github.com/jwplayer/jwplayer/commit/a20e4d2e4d858f41c59f1a54257709bffb287051 (rebase or merge in master to get the fix)

robwalch reviewed (changes requested):
1. Logo click-through is broken:
    - https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6009/allure/#suites/b9cb8385c37c0d6570a24dbf08ce345d/268763c48f174b2d/
    - http://player-pr-test-jenkins.longtailvideo.com/builds/28389/archive/test/harness.html?setup&edition=enterprise&setup&config=player_ui/logo-control-bar-svg.json
2. Found and fixed a regression in the test results with playlist loading https://github.com/jwplayer/jwplayer/commit/a20e4d2e4d858f41c59f1a54257709bffb287051 (rebase or merge in master to get the fix)

zetagame commented:
>     1. Logo click-through is broken:
>        
>        * https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6009/allure/#suites/b9cb8385c37c0d6570a24dbf08ce345d/268763c48f174b2d/
>        * http://player-pr-test-jenkins.longtailvideo.com/builds/28389/archive/test/harness.html?setup&edition=enterprise&setup&config=player_ui/logo-control-bar-svg.json
> 
>     2. Found and fixed a regression in the test results with playlist loading [a20e4d2](https://github.com/jwplayer/jwplayer/commit/a20e4d2e4d858f41c59f1a54257709bffb287051) (rebase or merge in master to get the fix)
> 
>     3. This PR [jwplayer/jwplayer-ads-vast#690](https://github.com/jwplayer/jwplayer-ads-vast/pull/690) fixes a regression in the vast plugin caused by earlier changes to the UI util. After that is merged rerun tests here.

I've added commits all action items above.

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1901/) MULTI Build for commit 33e56576652ccf7d78dd7da69c1f2da4c77e841b is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9324/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9324/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6268/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6268/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28394/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28394/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23845/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23845/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6012/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28394/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28394/archive/test/view/) Views

mamaddox reviewed (changes requested):
Looks like there is something going on with the playlist overlay where you can't click any of the tiles in it. Looking a bit further into it.
http://player-pr-test-jenkins.longtailvideo.com/builds/28394/archive/test/squash/bootstrap.html?primary=html5&edition=ads&feature=playlist/playlists_overlay&suite=playlists

jwrobot-player commented:
[:exclamation:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1903/) MULTI Build for commit d4549332400a3823fd3204ab8abdf9a6c45a6c6f did not complete (FAILURE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/none/consoleFull) jwplayer build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6270/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6270/consoleFull) jwplayer unit tests FAILURE
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/consoleFull) jwplayer-commercial build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/none/consoleFull) jwplayer-commercial unit tests NOT_STARTED
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/none/allure/) NOT_STARTED
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/view/) Views

robwalch commented:
> I've added commits all action items above.

You did not rebase or merge master into this branch:

<img width="1293" alt="Screen Shot 2021-06-08 at 10 42 43 AM" src="https://user-images.githubusercontent.com/333258/121206219-4a046500-c846-11eb-9134-3834097291d2.png">

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1904/) MULTI Build for commit af00a0e91ffd55fb9c9d0a2b78abd666b6befdac is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9327/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9327/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6271/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6271/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28396/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28396/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23847/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23847/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6013/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28396/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28396/archive/test/view/) Views

zetagame commented:
> > I've added commits all action items above.
> 
> You did not rebase or merge master into this branch:
> 
> <img alt="Screen Shot 2021-06-08 at 10 42 43 AM" width="1293" src="https://user-images.githubusercontent.com/333258/121206219-4a046500-c846-11eb-9134-3834097291d2.png">

Was referring to 

> > I've added commits all action items above.
> 
> You did not rebase or merge master into this branch:
> 
> <img alt="Screen Shot 2021-06-08 at 10 42 43 AM" width="1293" src="https://user-images.githubusercontent.com/333258/121206219-4a046500-c846-11eb-9134-3834097291d2.png">

Rebased, building now.

robwalch commented on src/js/view/utils/add-click-action.ts:
```diff
@@ -1,7 +1,7 @@
 
-export function addClickAction(element: HTMLElement, clickAction: (evt: Event) => void, ctx?: any): UI {
-    const ui = new UI(element).on('click', clickAction, ctx);
+export function addClickAction(element: HTMLElement, clickAction: (evt: Event) => void, ctx?: any, directSelect?: boolean): UI {
```
I don't see a forth `directSelect` option being passed in anywhere. Can you point out where this is being used. Otherwise I don't see the justification for adding it.

zetagame replied:
Removed. I was going to use this for volumeTooltipIcon/future use, but it conflicted with the enter handling since it's a group element.

jwrobot-player commented:
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1906/) MULTI Build for commit c46a6b617ccb5693defa508dcdcc1824f0c24fa4 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9329/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/9329/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6273/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6273/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28399/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/28399/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/23850/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/23850/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/6016/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/28399/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/28399/archive/test/view/) Views
